applications/FluidDynamicsHydraulicsApplication/python_scripts/apply_hydraulic_inlet_process.py

```diff
@@ -0,0 +1,181 @@
+        water_depth_b = max([node.Z for node in self.inlet_model_part.Nodes])
+        # The froude number for supercritical(small water depth) flows has a big value while for subcritical(High water depth) flows the froude number is small less than one.
+        froude_number_a = 100000
+        froude_number_c = 0.0
+
+        #  Bisection method.
+        iter_count = 0
+        while (abs(froude_number_c - 1) > self.bisection_tolerance) and iter_count < self.maximum_iterations:
+            water_depth_c = 0.5*(water_depth_a+water_depth_b)
+            froude_number_c = self.CalculateFroudeNumber(water_depth_c)
+            if (froude_number_c-1)*(froude_number_a-1)<0:
+                water_depth_b=water_depth_c
+            else:
+                water_depth_a=water_depth_c
+                froude_number_a = froude_number_c
+            iter_count +=1
+        if iter_count == self.maximum_iterations:
+            KratosMultiphysics.Logger.PrintWarning("::[KratosFluidHydraulics]::", "Maximum iteractions for calculating critical water depth have been exceeded.")
+
+        # Obtain the corresponding velocity according to the wetted area and the inlet discharge value.
+        inlet_velocity = self.inlet_discharge/self.wetted_area
+
+        # Assing the inlet velocity to inlet nodes and fix it.
+        # TODO: Create a variable in kratos por Inlet normal, now it the variable uses for that is DISPLACEMENT.
+        KratosMultiphysics.NormalCalculationUtils().CalculateOnSimplexNonHistorical(self.inlet_model_part,self.domain_size,KratosMultiphysics.DISPLACEMENT)
+        KratosFluidHydraulics.HydraulicFluidAuxiliaryUtilities.SetInletVelocity(
+            self.inlet_model_part, inlet_velocity, self.water_depth_variable)
+
+        #Assign the identical value of the inlet water depth to the DISTANCE variable (free surface) for all nodes associated with the inlet model part.
+        KratosFluidHydraulics.HydraulicFluidAuxiliaryUtilities.SetInletFreeSurface(
+            self.inlet_model_part, KratosMultiphysics.INLET, self.water_depth_variable)
+
+
+    def ExecuteFinalizeSolutionStep(self):
+        # Here we free all of the nodes in the inlet
+        if self.step_is_active:
+            KratosFluidHydraulics.HydraulicFluidAuxiliaryUtilities.FreeInletVelocity(
+                self.inlet_model_part)
+        self.step_is_active = False
+
+    def CalculateFroudeNumber(self, water_depth):
+        #The Froude number is computed, and this value will be included into the objective function for the bisection method.
+        gravity = 9.81
+
+        # Assing the inlet free surface
+        for node in self.inlet_model_part.Nodes:
+            aux_distance = node.Z-water_depth
+            if abs(aux_distance) < self.water_depth_tolerance:
+                aux_distance = self.water_depth_tolerance if aux_distance > 0.0 else - self.water_depth_tolerance
+            node.SetValue(self.water_depth_variable, aux_distance)
```

Comment:
Don't you have a method in the auxiliary utils for doing this?

Reply:
I can not find it, but I will check it again.